scripts/icu/63.1-min-static-data-1/script.sh

```diff
@@ -0,0 +1,150 @@
+
+MASON_NAME=icu
+MASON_VERSION=63.1-min-static-data-1
+MASON_LIB_FILE=lib/libicuuc.a
+#MASON_PKGCONFIG_FILE=lib/pkgconfig/icu-uc.pc
+
+. ${MASON_DIR}/mason.sh
+
+MASON_BUILD_DEBUG=0 # Enable to build library with debug symbols
+MASON_CROSS_BUILD=0
+
+function mason_load_source {
+    # you can't get the data and the code together in one place except when you clone the repo
+    export MASON_BUILD_PATH=${MASON_ROOT}/.build/${MASON_NAME}-${MASON_VERSION}
+    if [[ ! -d ${MASON_BUILD_PATH} ]]; then
+        git clone https://github.com/unicode-org/icu --depth=1 --branch release-63-1 ${MASON_BUILD_PATH}
+    fi
+}
+
+function mason_prepare_compile {
+    if [[ ${MASON_PLATFORM} == 'ios' || ${MASON_PLATFORM} == 'android' || ${MASON_PLATFORM_VERSION} != `uname -m` ]]; then
+        mason_substep "Cross-compiling ICU. Starting with host build of ICU to generate tools."
+
+        pushd ${MASON_ROOT}/..
+        env -i HOME="$HOME" PATH="$PATH" USER="$USER" ${MASON_DIR}/mason build ${MASON_NAME} ${MASON_VERSION}
+        popd
+
+        # TODO: Copies a bunch of files to a kind of orphaned place, do we need to do something to clean up after the build?
+        #  Copying the whole build directory is the easiest way to do a cross build, but we could limit this to a small subset of files (icucross.mk, the tools directory, probably a few others...)
+        #  Also instead of using the regular build steps, we could use a dedicated built target that just builds the tools
+        mason_substep "Moving host ICU build directory to ${MASON_ROOT}/.build/icu-host"
+        rm -rf ${MASON_ROOT}/.build/icu-host
+        cp -R ${MASON_BUILD_PATH}/icu4c/source ${MASON_ROOT}/.build/icu-host
+    fi
+}
+
+function trim_data {
+    mason_substep "Trimming ICU data to small number of locals and datasets"
+
+    # for some reason there is also a dat file in there!
+    rm -f data/in/icudt63l.dat
+
+    # move all non algorithmic code point mappings so they arent used
+    for f in $(find data/mappings -name '*.mk'); do
+        mv $f ${f}_
+    done
+
+    # make local versions of each data set
+    for f in $(find data -path mappings -prune -o -name '*files.mk'); do
+        # make a local version and remove the line continuations
+        l=$(echo $f | sed -e "s/\(.*\)files.mk/\1local.mk/g")
+        sed -e :a -e '/\\$/N; s/\\\n//; ta' $f > $l
+
+        # if its locale, unit or currency we use our supported lang list
+        if [ "$(echo $l | grep -cE 'locales|unit|curr')" -eq 1 ]; then
+            sed -i'' -e '/^#/!s/SOURCE.*=.*txt/SOURCE = da.txt de.txt en.txt eo.txt es.txt fi.txt fr.txt he.txt id.txt it.txt ko.txt my.txt nl.txt pl.txt pt.txt pt_PT.txt ro.txt ru.txt sv.txt tr.txt uk.txt vi.txt zh.txt zh_Hans.txt/g' $l
+        # if its misc we need a couple of things
+        elif [ $(echo $l | grep -cF "misc") -eq 1 ]; then
+            sed -i'' -e '/^#/!s/SOURCE.*=.*txt/SOURCE = plurals.txt numberingSystems.txt icuver.txt icustd.txt pluralRanges.txt/g' $l
```

Comment:
this is the only other difference in this package. we've added `plurals.txt` to the list of data to keep in the library. it turns out this is required to get the library to spit out `1 mile` instead of `1 miles`. :man_facepalming:

Reply:
wow, nice work figuring this out 🙌